packages/node-core/src/indexer/dictionary/dictionary.service.spec.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020-2024 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import {EventEmitter2} from '@nestjs/event-emitter';
+import {NETWORK_FAMILY} from '@subql/common';
+import {NodeConfig} from '../..';
+import {DictionaryService} from './dictionary.service';
+import {TestDictionaryV1} from './v1/dictionaryV1.test';
+import {TestDictionaryV2, TestFB} from './v2/dictionaryV2.test';
+
+class TestDictionaryService extends DictionaryService<any, TestFB> {
+  async initDictionaries(): Promise<void> {
+    // Mock version inspection completed
+    const dictionaryV1Endpoints = [
+      'https://gx.api.subquery.network/sq/subquery/eth-dictionary',
+      'https://dict-tyk.subquery.network/query/eth-mainnet',
+    ];
+    const dictionaryV2Endpoints = ['http://localhost:3000/rpc'];
+
+    const dictionariesV1 = await Promise.all(
+      dictionaryV1Endpoints.map(
+        (endpoint) => new TestDictionaryV1(endpoint, 'mockChainId', this.nodeConfig, this.eventEmitter)
+      )
+    );
+    const dictionariesV2 = dictionaryV2Endpoints.map(
+      (endpoint) => new TestDictionaryV2(endpoint, 'mockChainId', this.nodeConfig, this.eventEmitter)
+    );
+    this.init([...dictionariesV1, ...dictionariesV2]);
+  }
+}
+
+describe('Dictionary service', function () {
+  let dictionaryService: TestDictionaryService;
+
+  beforeEach(async () => {
+    const nodeConfig = new NodeConfig({
+      subquery: 'dictionaryService',
+      subqueryName: 'asdf',
+      networkEndpoint: ['wss://eth.api.onfinality.io/public-ws'],
+      dictionaryTimeout: 10,
+      dictionaryResolver: false,
+      networkDictionary: [
+        'https://gx.api.subquery.network/sq/subquery/eth-dictionary',
+        'https://dict-tyk.subquery.network/query/eth-mainnet',
+        'http://localhost:3000/rpc',
```

Comment:
I don't think these tests should be dependent on running other things locally. Unless we run the services for docker integration tests